### DISCUSSION TITLE
Unable to run SpineOpt using Gurobi

### DISCUSSION THREAD

**Discussion by Konathanasiou** [spine-tools/SpineOpt.jl#474] [Support]
Hi everyone,

I've been trying to run SpineOpt using Gurobi solver and I'm getting the following error:

ERROR: LoadError: MethodError: no method matching run_spineopt(; mip_solver=MathOptInterface.OptimizerWithAttributes(Gurobi.Optimizer, Pair{MathOptInterface.AbstractOptimizerAttribute, Any}[]), lp_solver=MathOptInterface.OptimizerWithAttributes(Gurobi.Optimizer, Pair{MathOptInterface.AbstractOptimizerAttribute, Any}[]))run_spineopt(!Matched::String, !Matched::String; upgrade, mip_solver, lp_solver, cleanup, add_user_variables, add_constraints, update_constraints, log_level, optimize, use_direct_model) at C:\ProgramData\NEMO\depot\packages\SpineOpt\AMGEb\src\run_spineopt.jl:81
Stacktrace:
[1] top-level scope

Closest candidates are:run_spineopt(!Matched::String) at C:\ProgramData\NEMO\depot\packages\SpineOpt\AMGEb\src\run_spineopt.jl:81 got unsupported keyword arguments "mip_solver", "lp_solver"

@ C:\Users\Nosdi\.spinetoolbox\plugins\SpineOpt\specifications\Tool\run_spineopt.jl:5
[2] include(fname::String)
@ Base.MainInclude .\client.jl:451
[3] top-level scope
@ none:1
in expression starting at C:\Users\Nosdi\.spinetoolbox\plugins\SpineOpt\specifications\Tool\run_spineopt.jl:5

My run_spineopt.jl file looks like this (the one located at C:\Users\Nosdi\.spinetoolbox\plugins\SpineOpt\specifications\Tool) :
![image](https://user-images.githubusercontent.com/102978432/177433295-2962e5b4-d198-4006-900e-25493f26438c.png)

However, the error is coming from a file also called run_spineopt.jl in a different location (C:\ProgramData\NEMO\depot\packages\SpineOpt\AMGEb\src\run_spineopt.jl:81) which is an automatically generated file when downloading SpineOpt apparently.

Any ideas on how to solve this issue and make Gurobi work? I've used Gurobi to solve other LP and MIP models before, so there's nothing wrong with the solver itself.

Thank you in advance!

**Comment by manuelma** [comment]
Hi! thanks very much for trying SpineOpt.

I'm pretty sure your issue comes from running SpineOpt from toolbox without passing the connected Data Store URLs as Tool arguments. Checkout my screenshot below. On the right, you see the Tool properties. See under Tool 'arguments', the two db urls I'm using to run SpineOpt (input and output). You can achieve the same by dragging and dropping the urls from 'Available resources' below. Note that the order is important! The first url under 'Tool arguments' needs to be the input, and the second needs to be the output. Please let me know how it goes :)

![image](https://user-images.githubusercontent.com/11472756/177483034-9d40225e-585d-4d4c-b035-8efc1e254f14.png)

**Comment by Konathanasiou** [comment]
I think you're absolutely right, I don't know how I missed that to be honest, thank you so much!

**Comment by Konathanasiou** [comment]
Do you maybe also know why my environments base file is within Nemo even though I'm not using it? I uninstalled all my packages when deleted manually the whole Nemo file thinking that this is the problem but when starting installing them again they would automatically be generated at this location 'C:\ProgramData\NEMO\depot\packages\SpineOpt' even though I'm not using NEMO.

**Reply by mihlema** [reply]
Hi @Konathanasiou,
I think this might be related to this [issue](https://discourse.julialang.org/t/julia-adding-an-older-version-of-my-updated-gurobi-solber/74034/15). Did you happen to have installed the julia version from NEMO? I would recommend uninstalling NEMO and installing julia from [here](https://julialang.org/downloads/).
